types/src/proof/definition.rs

```diff
@@ -335,7 +337,10 @@ impl TransactionAccumulatorSummary {
         consistency_proof: &AccumulatorConsistencyProof,
         target_li: &LedgerInfo,
     ) -> Result<Self> {
-        let num_new_txns = target_li.version() - self.0.version();
+        let num_new_txns = target_li
+            .version()
+            .checked_sub(self.0.version())
+            .ok_or_else(|| format_err!("underflow"))?;
```

Comment:
I think this is more of a "target_li older than my version" type of logic error that we want to be explicit.